pkg/ibmvpc/vpc.go

```diff
@@ -324,15 +378,15 @@ func (pgw *PublicGateway) ExternalIP() string {
 
 func (pgw *PublicGateway) AllowedConnectivity(src, dst vpcmodel.VPCResourceIntf) (*netset.TransportSet, error) {
 	if areNodes, src1, dst1 := isNodesPair(src, dst); areNodes {
-		if vpcmodel.HasNode(pgw.Sources(), src1) && dst1.IsExternal() {
+		if vpcmodel.HasNode(pgw.Sources(), src1) && dst1.IsExternal() && dst1.IsPublicInternet() {
```

Comment:
A similar change is also required in `func (fip *FloatingIP) AllowedConnectivity()` and in `func (fip *FloatingIP) RouterDefined()`